core/src/actors/chain_manager/mod.rs

```diff
@@ -226,15 +228,28 @@ impl ChainManager {
     fn build_block(&self, msg: &BuildBlock) -> Block {
         // Get all the unspent transactions and calculate the sum of their fees
         let mut transaction_fees = 0;
-        let transactions: Vec<Transaction> = self
-            .transactions_pool
-            .iter()
-            .map(|t| {
-                // TODO: t.fee()
-                transaction_fees += 1;
-                *t
-            })
-            .collect();
+        let max_block_weight = self.max_block_weight;
+        let mut block_weight = 0;
+        let mut transactions = Vec::new();
+
+        for transaction in self.transactions_pool.iter() {
```

Comment:
Not functional enough

Reply:
Nah, the thing is that I added a small optimization: if the weight of the included transactions is equal to the maximum weight allowed for the block, then there's no need to keep processing more transactions so a `break` is used to stop the loop.